src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordEngine.cs

```diff
@@ -394,6 +446,75 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
             }));
         }
 
+        private static async Task RefactorInitializersAsync(
+            INamedTypeSymbol type,
+            SolutionEditor solutionEditor,
+            ImmutableArray<IPropertySymbol> positionalParameters,
+            CancellationToken cancellationToken)
+        {
+            var symbolReferences = await SymbolFinder
+                .FindReferencesAsync(type, solutionEditor.OriginalSolution, cancellationToken).ConfigureAwait(false);
+            var referenceLocations = symbolReferences.SelectMany(reference => reference.Locations);
+            var documentLookup = referenceLocations.ToLookup(refLoc => refLoc.Document.Id);
+            foreach (var (documentID, documentLocations) in documentLookup)
+            {
+                var documentEditor = await solutionEditor
+                        .GetDocumentEditorAsync(documentID, cancellationToken).ConfigureAwait(false);
+                if (documentEditor.OriginalDocument.Project.Language != LanguageNames.CSharp)
+                {
+                    // since this is a CSharp-dependent file, we need to have specific VB support.
+                    // for now skip VB usages.
+                    continue;
+                }
+
+                var objectCreationExpressions = documentLocations
+                    // we should find the identifier node of an object creation expression
+                    .Select(referenceLocations => referenceLocations.Location.FindNode(cancellationToken).Parent)
+                    .OfType<ObjectCreationExpressionSyntax>();
+                foreach (var objectCreationExpression in objectCreationExpressions)
+                {
+                    if (documentEditor.SemanticModel.GetOperation(objectCreationExpression, cancellationToken)
+                        is not IObjectCreationOperation objectCreationOperation)
```

Comment:
Consider a stronger Debug.Assert or something here; if this is giving us a different IOperation that seems fishy.